src/OSPSuite.UI/Services/DialogCreator.cs

```diff
@@ -80,26 +83,44 @@ public ViewResult MessageBoxYesNo(string message, ViewResult defaultButton = Vie
       public ViewResult MessageBoxYesNo(string message, string yes, string no, ViewResult defaultButton = ViewResult.Yes)
       {
          return showQuestionMessageBox(message, new [] {
-            DialogResult.Yes, DialogResult.No}, yes, no, string.Empty, defaultButton);
+            DialogResult.Yes, DialogResult.No}, yes, no, string.Empty, defaultButton, null);
       }
 
-      private ViewResult showQuestionMessageBox(string message, IReadOnlyList<DialogResult> buttons, string yes, string no, string cancel, ViewResult defaultButton)
+      public ViewResult MessageBoxConfirm(string message, Action doNotShowAgain, ViewResult defaultButton = ViewResult.Yes)
+      {
+         return showQuestionMessageBox(message, new[] {
+            DialogResult.Yes, DialogResult.Cancel}, string.Empty, string.Empty, string.Empty, defaultButton, doNotShowAgain);
+      }
+
+      private ViewResult showQuestionMessageBox(string message, IReadOnlyList<DialogResult> buttons, string yes, string no, string cancel, ViewResult defaultButton, Action doNotShowAgain)
       {
          var currentLocalizer = Localizer.Active;
          try
          {
             Localizer.Active = new XtraMessageBoxLocalizer(yes, no, cancel);
-            return _mapper.MapFrom(XtraMessageBox.Show(createMessageBoxArgs(message, buttons.ToArray(), getIcon(SystemIcons.Question), defaultButtonFrom(defaultButton))));
+            var args = createMessageBoxArgs(message, buttons.ToArray(), getIcon(SystemIcons.Question), defaultButtonFrom(defaultButton), doNotShowAgain != null);
+            args.Closed += (sender, e) => onMessageBoxClosed(e, doNotShowAgain);
+            DialogResult input = XtraMessageBox.Show(args);
+            return _mapper.MapFrom(input);
          }
          finally
          {
             Localizer.Active = currentLocalizer;
          }
       }
 
+      private void onMessageBoxClosed(XtraMessageBoxClosedArgs e, Action doNotShowAgain)
+      {
+          // e.Visible gets whether the user checked the 'Do not show this message again' checkbox. Checked makes e.Visible = false 
+          if (!e.Visible && e.DialogResult == DialogResult.Yes)
+          {
+              doNotShowAgain?.Invoke();
```

Comment:
because here, is where it's used, and it's checked for ```null``` first

Reply:
You're right. Fixed in commit 1ad367850ff74c4378df84d90be4dac856d619db